jscomp/ml/ast_untagged_variants.ml

```diff
@@ -341,9 +356,8 @@ module DynamicChecks = struct
   let add_runtime_type_check ~tag_type ~(block_cases: block_type list) x y =
     let has_array() = Ext_list.exists block_cases (fun t -> t = ArrayType) in
     match tag_type with
-    | Untagged IntType
-    | Untagged StringType
-    | Untagged FloatType -> typeof y == x
```

Comment:
This seems OK as functions can't overlap with int, string, float, object, array.